management/src/main/scala/akka/management/javadsl/HealthChecks.scala

```diff
@@ -27,19 +26,31 @@ final class HealthChecks(system: ExtendedActorSystem, settings: HealthCheckSetti
 
   private val delegate = new HealthChecksImpl(system, settings)
 
+  /**
+   * Returns CompletionStage(result), containing the system's readiness result
+   */
+  def readyResult(): CompletionStage[CheckResult] =
+    delegate.readyResult().map(new CheckResult(_))(system.dispatcher).toJava
+
   /**
    * Returns CompletionStage(true) if the system is ready to receive user traffic
    */
   def ready(): CompletionStage[java.lang.Boolean] =
-    delegate.ready().map(Boolean.box)(ExecutionContexts.sameThreadExecutionContext).toJava
+    readyResult().thenApply(((r: CheckResult) => r.isSuccess).asJava)
```

Comment:
Originally had `.thenApply(_.iSuccess).asJava` but that failed to compile on 2.11:

```
[error] /Users/jlongshore/work/tubi/akka-management/management/src/main/scala/akka/management/javadsl/HealthChecks.scala:38:29: missing parameter type for expanded function ((x$2) => x$2.isSuccess)
[error]     readyResult().thenApply(_.isSuccess)
[error]                             ^
[error] /Users/jlongshore/work/tubi/akka-management/management/src/main/scala/akka/management/javadsl/HealthChecks.scala:52:29: missing parameter type for expanded function ((x$4) => x$4.isSuccess)
[error]     aliveResult().thenApply(_.isSuccess)
[error]                             ^
[error] two errors found
```